File: data structures and algorithms/python/08_035.py
def max_property_profit(prices):

    if not prices or len(prices) < 2:
        return 0
    
    min_price = float('inf')
    max_profit = 0
    
    for price in prices:
        if price < min_price:
            min_price = price
        else:
            current_profit = price - min_price
            max_profit = max(max_profit, current_profit)
    
    return max_profit

File: data structures and algorithms/cs/08_035.cs
using System;

public class PropertyProfitCalculator
{
   public static int MaxPropertyProfit(int[] prices)
   {
      if (prices == null || prices.Length < 2)
      {
         return 0;
      }

      int minPrice = int.MaxValue;
      int maxProfit = 0;

      foreach (int price in prices)
      {
         if (price < minPrice)
         {
            minPrice = price;
         }
         else
         {
            int currentProfit = price - minPrice;
            maxProfit = Math.Max(maxProfit, currentProfit);
         }
      }

      return maxProfit;
   }
}